rapid/src/main/java/com/vrg/rapid/MembershipService.java

```diff
@@ -494,9 +509,36 @@ int getMembershipSize() {
      * Shuts down all the executors.
      */
     void shutdown() {
+        leave();
         alertBatcherJob.cancel(true);
         failureDetectorJobs.forEach(k -> k.cancel(true));
         messagingClient.shutdown();
+        membershipView.reset();
+    }
+
+    /**
+     * Leaves the cluster by telling all the observers to proactively trigger failure.
+     * This operation is blocking, as we need to wait to send the alert messages before shutting down the rest
+     */
+    void leave() {
+        final LeaveMessage leaveMessage = LeaveMessage.newBuilder().setSender(myAddr).build();
+        final RapidRequest leave = RapidRequest.newBuilder().setLeaveMessage(leaveMessage).build();
+        try {
+            membershipView.getObserversOf(myAddr).forEach(endpoint -> {
```

Comment:
You can use the Futures.**List methods to run them in parallel and wait on the results of all of them. https://guava.dev/releases/snapshot/api/docs/com/google/common/util/concurrent/Futures.html#allAsList-java.lang.Iterable-